src/devices/cpu/hd61700/hd61700.cpp

```diff
@@ -814,27 +794,19 @@ void hd61700_cpu_device::execute_run()
 						uint8_t arg = read_op();
 						REG_IZ += get_sign_mreg(arg);
 
-						WRITE_REG(arg, mem_readbyte(REG_UA>>6, REG_IZ));
+						WRITE_REG(arg, mem_readbyte(REG_UA >> 6, REG_IZ));
 
 						m_icount -= 6;
 					}
 					break;
 
 				case 0x2e:  //pps
-					{
-						uint8_t arg = read_op();
-						WRITE_REG(arg, pop(REG_SS));
-
-						m_icount -= 11;
-					}
+					WRITE_REG(read_op(), pop(REG_SS));
+					m_icount -= 11;
 					break;
 				case 0x2f:  //ppu
-					{
-						uint8_t arg = read_op();
-						WRITE_REG(arg, pop(REG_US));
-
-						m_icount -= 11;
-					}
+					WRITE_REG(read_op(), pop(REG_US));
+					m_icount -= 11;
```

Comment:
This isn’t safe – the order of evaluation of the arguments passed to the macro isn’t guaranteed, so the side effects of the `pop` may happen before the `read_op`.  You need the intermediate local to ensure things happen in the right order.

Reply:
Fixed.